src/components/Form/FormWrapper.tsx

```diff
@@ -0,0 +1,171 @@
+                                // consistent results across mobile and web
+                                // eslint-disable-next-line @typescript-eslint/naming-convention
+                                focusInput?.measureLayout?.(formContentRef.current, (_x, y) =>
+                                    formRef.current?.scrollTo({
+                                        y: y - 10,
+                                        animated: false,
+                                    }),
+                                );
+                            }
+
+                            // Focus the input after scrolling, as on the Web it gives a slightly better visual result
+                            focusInput?.focus?.();
+                        }}
+                        containerStyles={[styles.mh0, styles.mt5, styles.flex1, submitButtonStyles]}
+                        enabledWhenOffline={enabledWhenOffline}
+                        isSubmitActionDangerous={isSubmitActionDangerous}
+                        disablePressOnEnter
+                    />
+                )}
+            </FormSubmit>
+        ),
+        [
+            children,
+            enabledWhenOffline,
+            errorMessage,
+            errors,
+            footerContent,
+            formID,
+            formState?.errorFields,
+            formState?.isLoading,
+            inputRefs,
+            isSubmitActionDangerous,
+            isSubmitButtonVisible,
+            onSubmit,
+            style,
+            styles.flex1,
+            styles.mh0,
+            styles.mt5,
+            submitButtonStyles,
+            submitButtonText,
+        ],
+    );
+
+    return (
+        <SafeAreaConsumer>
+            {({safeAreaPaddingBottomStyle}) =>
+                scrollContextEnabled ? (
+                    <ScrollViewWithContext
+                        style={[styles.w100, styles.flex1]}
+                        contentContainerStyle={styles.flexGrow1}
+                        keyboardShouldPersistTaps="handled"
+                        ref={formRef}
+                    >
+                        {scrollViewContent(safeAreaPaddingBottomStyle)}
+                    </ScrollViewWithContext>
+                ) : (
+                    <ScrollView
+                        style={[styles.w100, styles.flex1]}
+                        contentContainerStyle={styles.flexGrow1}
+                        keyboardShouldPersistTaps="handled"
+                        ref={formRef}
+                    >
+                        {scrollViewContent(safeAreaPaddingBottomStyle)}
+                    </ScrollView>
+                )
+            }
+        </SafeAreaConsumer>
+    );
+}
+
+FormWrapper.displayName = 'FormWrapper';
+
+export default withOnyx<FormWrapperProps, FormWrapperOnyxProps>({
+    formState: {
+        // FIX: Fabio plz help 😂
+        key: (props) => props.formID as typeof ONYXKEYS.FORMS.EDIT_TASK_FORM,
+    },
```

Comment:
Hey @fabioh8010 could you please take look here and in [similar place in FormProvider](https://github.com/Expensify/App/pull/32992/files#diff-16b60eed4ca05874fc3fadcb18fc790efcb94648af3991004dfa67c7e7f39923R343-R353)? We've been struggling with @blazejkustra but we are run out of ideas how to type it.

Reply:
@kowczarz @blazejkustra  `withOnyx` typings are not able to handle such generic cases like this one, since it's a generic component I think we can just cast it to `any` and leave a comment.

```ts
export default withOnyx<FormWrapperProps, FormWrapperOnyxProps>({
    formState: {
        // Comment explaining..
        // eslint-disable-next-line @typescript-eslint/no-unsafe-return, @typescript-eslint/no-explicit-any
        key: (props) => props.formID as any,
    },
})(FormWrapper);
```

WDYT?